File: api/Models/LoyaltyDiscountTrackerReport.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ByzmoApi.Models
{
    public class LoyaltyDiscountTrackerReport
    {
        public string Name { get; set; }
        public string Email { get; set; }
        public int UserId { get; set; }
        public string DiscountCode { get; set; }
        public string TierLevel { get; set; }
        public decimal AccumulatedPurchaseAmount { get; set; }
        public decimal LoyaltyId { get; set; }
        public IEnumerable<LoyaltyDiscountTracker> loyaltyDiscountTracker { get; set; }
    }
}

File: web/src/app/classes/loyalty-discount-tracker-report.ts
import { LoyaltyDiscountTracker } from './loyalty-discount-tracker';

export class LoyaltyDiscountTrackerReport {
    name: string;
    email: string;
    userId: number;
    discountCode: string;
    tierLevel: string;
    accumulatedPurchaseAmount: number;
    loyaltyId: number;
    loyaltyDiscountTracker: Array<LoyaltyDiscountTracker>;

    constructor() {
        this.loyaltyDiscountTracker = new Array();
    }

}


